src/Service.Tests/CosmosTests/QueryTests.cs

```diff
@@ -472,6 +479,158 @@ ... p
+            {
+                $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string mutation = @"
+mutation ($id: ID!, $partitionKeyValue: String!, $item: UpdatePlanetInput!) {
+    updatePlanet (id: $id, _partitionKeyValue: $partitionKeyValue, item: $item) {
+        id
+        name
+     }
+}";
+
+                string graphQLQuery = @$"
+query {{
+    planet_by_pk (id: ""{id}"") {{
+        id
+        name
+    }}
+}}";
+                string queryName = "planet_by_pk";
+
+                // First query
+                JsonElement queryResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                            client,
+                                            server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                            query: graphQLQuery,
+                                            queryName: queryName
+                                            );
+                name = queryResponse.GetProperty("name").GetString();
+
+                var update = new
+                {
+                    id = id,
+                    name = "new_name",
+                };
+
+                // Mutated name property
+                JsonElement mutationResponse = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                client,
+                                server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                query: mutation,
+                                queryName: "updatePlanet",
+                                variables: new() { { "id", id }, { "partitionKeyValue", id }, { "item", update } },
+                                authToken: AuthTestHelper.CreateStaticWebAppsEasyAuthToken(),
+                                clientRoleHeader: AuthorizationResolver.ROLE_AUTHENTICATED
+                                );
+
+                // Asserting name is mutated
+                Assert.IsFalse(queryResponse.GetProperty("name").GetString().Equals(update.name), "Mutation didn't change the name successfully");
+
+                // Second query - the data returned is from the cache, not the mutated one
+                JsonElement queryResponse2 = await GraphQLRequestExecutor.PostGraphQLRequestAsync(
+                                             client,
+                                             server.Services.GetRequiredService<RuntimeConfigProvider>(),
+                                             query: graphQLQuery,
+                                             queryName: queryName
+                                             );
+
+                // Asserting cached data is returned
+                Assert.IsFalse(queryResponse.GetProperty("name").GetString() != name, "Query didn't return cached value");
```

Comment:
nit: Instead of negation, Asserting with IsTrue improves the readability of the test. 
Are we testing that old value is returned from the cache even after updating the name?

Reply:
Yes, we are testing the same cached value is returned and not the modified one. 
Negative assertion is added to provide a clear error message when the expected result doesn't match.